### DISCUSSION TITLE
Is that possible to manual bind module to client.

### DISCUSSION THREAD

**Discussion by xmeow** [microsoft/devicescript#661] [Q&A]
In some cases, we have multiple modules with the same service, but we want to bind different modules to specific clients.  So it won't be messy if we bind the module to the client by the short id.

Something like this:

```typescript
const btnLeft = new ds.Button("QK87")
const btnRight = new ds.Button("AU12")


```

**Comment by pelikhan** [comment]
@mmoskal binding is in jacdac-c?

**Reply by mmoskal** [reply]
Yeah. Would need to update role manager C impl. @xmeow note that binding is stable - alphabetic order of role names and number order of device IDs.

**Reply by mmoskal** [reply]
to be clear, you could call your buttons "0 Red" and "1 Blue" (assuming they are red and blue). If it doesn't match you can rename them "1 Red" and "0 Blue" - the assignment will be stable.

**Comment by xmeow** [comment]
Thank you guys. I see the problem, in the makefaire show people keeping plugin and remove modules and seems that module assignment is random.
